src/sentry/workflow_engine/handlers/detector/stateful.py

```diff
@@ -209,9 +211,29 @@ def evaluate_group_key_value(
                     new_substatus=None,
                 )
             else:
-                result, event_data = self.build_occurrence_and_event_data(
-                    group_key, value, PriorityLevel(new_status)
+                detector_occurrence, event_data = self.build_occurrence_and_event_data(
```

Comment:
🤔 i feel like this is getting a bit complex for this method -- can you see any places we could decompose this a little more? (it might also just be that other refactor i was talking about to make this a composed method)

Reply:
I think it's probably part of the other refactor - or might just be easier to work on as part of that work